### PR TITLE
Updates dapr/dapr to v1.12

### PR DESCRIPTION
**Pull request by JoshVanL**
Updates dapr/dapr to `1.12`

*PR currently points to the unreleased tag [v1.12.0-rc2](https://github.com/dapr/dapr/releases/tag/v1.12.0-rc.2)*

/cc @mukundansundar

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,248 +3,243 @@ module github.com/dapr/cli
...
+	github.com/spf13/viper v1.16.0
+	github.com/stretchr/testify v1.8.4
+	golang.org/x/sys v0.12.0
 	gopkg.in/yaml.v2 v2.4.0
-	helm.sh/helm/v3 v3.11.1
-	k8s.io/api v0.26.3
-	k8s.io/apiextensions-apiserver v0.26.3
-	k8s.io/apimachinery v0.26.3
-	k8s.io/cli-runtime v0.26.3
-	k8s.io/client-go v0.26.3
```
Comment (mukundansundar) [inline go.mod:33]: We still want to support 26.2 for this release can be updated in the next release . There is an issue tracking it.

**Review by mukundansundar — CHANGES_REQUESTED** [review]

**Comment by pravinpushkar** [comment]
Should we merge the #1331 to this PR ?

File: go.mod (hunk 1 above)
Comment (pravinpushkar) [reply go.mod]: I have pushed a commit to update the runtime to rc.2 and have reverted the k8s package update in this commit - https://github.com/dapr/cli/pull/1342/commits/a139d396ff404ffd83032c4f6e4245604dfac3cb

let me know what you think ? @JoshVanL @mukundansundar

Comment (JoshVanL) [reply go.mod]: @pravinpushkar please try not to push to authors branches, and instead favor code review or PRing to their branch :slightly_smiling_face:

Comment (pravinpushkar) [reply go.mod]: @JoshVanL Sure, thought you will not be able to look into it until my night time. So, was only trying to fast pace the process for RC cut.

**Review by artursouza — APPROVED** [review]

File: cmd/renew_certificate.go [hunk 2]
```diff
@@ -168,22 +170,43 @@ func logErrorAndExit(err error) {
...
+			}
+		}(i, name)
+	}
+
+	wg.Wait()
+
+	if err := errors.Join(errs...); err != nil {
+		return err
 	}
+
```
Comment (JoshVanL) [inline cmd/renew_certificate.go:209]: Sentry and injector have a codependency so need to be rolled out at the same time. Control plane upgrades are considered atomic anyway, so we do all services at the same time.

**Review by shubham1172 — APPROVED** [review]

**Review by pravinpushkar — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**